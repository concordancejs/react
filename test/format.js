import test from 'ava'
import concordance from 'concordance'
import React from 'react'
import renderer from 'react-test-renderer'

import plugin from '..'
import HelloMessage from './fixtures/react/HelloMessage'

const plugins = [plugin]
const format = (value, options) => concordance.format(value, Object.assign({ plugins }, options))

const snapshot = (t, getValue, options) => {
  t.snapshot(format(getValue(), options))
}
snapshot.title = prefix => `formats ${prefix}`
const snapshotRendered = (t, getValue, options) => {
  const tree = renderer.create(getValue()).toJSON()
  t.snapshot(format(tree, options))
}
snapshotRendered.title = prefix => `formats rendered ${prefix}`

const macros = [snapshot, snapshotRendered]

test('react elements', macros, () => <HelloMessage name='John' />)
test('fragments', macros, () => <React.Fragment><HelloMessage name='John' /></React.Fragment>)
test('object properties', macros, () => {
  return React.createElement('Foo', { object: { baz: 'thud' } })
})
test('array values in object properties', macros, () => {
  return React.createElement('Foo', { object: { baz: ['thud'] } })
})
test('array values in object properties in children', macros, () => {
  return React.createElement('Foo', null, React.createElement('Bar', { key: 'bar', object: { baz: ['thud'] } }))
})
test('multiline string properties', macros, () => {
  return React.createElement('Foo', { multiline: 'foo\nbar' })
})
test('illegal spaces in property names', macros, () => {
  return React.createElement('Foo', { 'foo bar': 'baz' })
})
test('string and number children', macros, () => {
  return React.createElement('div', null, 'foo', 'bar', 42)
})
test('properties and children', macros, () => {
  return React.createElement('Foo', { foo: 'bar' }, 'baz')
})

test('max depth', macros, () => {
  return (
    <div>
      <div>
        <div>
          Hello
        </div>
        <div id='foo'>Hello</div>
        <br id='bar'/>
        <br/>
      </div>
    </div>
  )
}, { maxDepth: 2 })

test('single line attribute values', snapshot, () => {
  return React.createElement('Foo', { bool: true })
})

test('opaque children', snapshot, () => {
  return (
    <div>
      {true}
      {-0}
      {() => {}}
      {new Set(['foo'])}
    </div>
  )
})
