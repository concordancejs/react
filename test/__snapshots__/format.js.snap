// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formats array values in object properties 1`] = `
"<Foo
  object={
    Object {
      baz: Array [
        'thud',
      ],
    }
  }
/>"
`;

exports[`formats array values in object properties in children 1`] = `
"<Foo>
  <Bar
    key=\\"bar\\"
    object={
      Object {
        baz: Array [
          'thud',
        ],
      }
    }
  />
</Foo>"
`;

exports[`formats illegal spaces in property names 1`] = `
"<Foo
  \\"foo bar\\"=\\"baz\\"
/>"
`;

exports[`formats max depth 1`] = `
"<div>
  <div>
    <div>…</div>
    <div…>…</div>
    <br… />
    <br />
  </div>
</div>"
`;

exports[`formats multiline string properties 1`] = `
"<Foo
  multiline={
    \`foo␊
    bar\`
  }
/>"
`;

exports[`formats object properties 1`] = `
"<Foo
  object={
    Object {
      baz: 'thud',
    }
  }
/>"
`;

exports[`formats opaque children 1`] = `
"<div>
  {true}
  {-0}
  {Function {}}
  {
    Set {
      'foo',
    }
  }
</div>"
`;

exports[`formats properties and children 1`] = `
"<Foo
  foo=\\"bar\\"
>
  baz
</Foo>"
`;

exports[`formats react elements 1`] = `
"<HelloMessage⍟
  name=\\"John\\"
/>"
`;

exports[`formats rendered array values in object properties 1`] = `
"<Foo
  object={
    Object {
      baz: Array [
        'thud',
      ],
    }
  }
/>"
`;

exports[`formats rendered array values in object properties in children 1`] = `
"<Foo>
  <Bar
    object={
      Object {
        baz: Array [
          'thud',
        ],
      }
    }
  />
</Foo>"
`;

exports[`formats rendered illegal spaces in property names 1`] = `
"<Foo
  \\"foo bar\\"=\\"baz\\"
/>"
`;

exports[`formats rendered max depth 1`] = `
"<div>
  <div>
    <div>…</div>
    <div…>…</div>
    <br… />
    <br />
  </div>
</div>"
`;

exports[`formats rendered multiline string properties 1`] = `
"<Foo
  multiline={
    \`foo␊
    bar\`
  }
/>"
`;

exports[`formats rendered object properties 1`] = `
"<Foo
  object={
    Object {
      baz: 'thud',
    }
  }
/>"
`;

exports[`formats rendered properties and children 1`] = `
"<Foo
  foo=\\"bar\\"
>
  baz
</Foo>"
`;

exports[`formats rendered react elements 1`] = `
"<div>
  Hello 
  <mark>
    John
  </mark>
</div>"
`;

exports[`formats rendered string and number children 1`] = `
"<div>
  foo
  bar
  {42}
</div>"
`;

exports[`formats single line attribute values 1`] = `
"<Foo
  bool={true}
/>"
`;

exports[`formats string and number children 1`] = `
"<div>
  foo
  bar
  {42}
</div>"
`;
